core/chains/evm/client/evm_client_test.go

```diff
@@ -40,3 +50,81 @@ func TestNewEvmClient(t *testing.T) {
 	client := client.NewEvmClient(nodePool, chainCfg, nil, logger.Test(t), testutils.FixtureChainID, nodes, chaintype.ChainType(chainTypeStr))
 	require.NotNil(t, client)
 }
+
+func TestChainClientMetrics(t *testing.T) {
+	ctx, cancel := context.WithTimeout(tests.Context(t), tests.WaitTimeout(t))
+	defer cancel()
+
+	nodeConfigs := []client.NodeConfig{
+		{
+			Name:    ptr("BlueEVMPrimaryNode"),
+			WSURL:   ptr("ws://no-blue-node"),
+			HTTPURL: ptr("http://no-blue-node"),
+		},
+		{
+			Name:    ptr("YellowEVMPrimaryNode"),
+			WSURL:   ptr("ws://no-yellow-node"),
+			HTTPURL: ptr("http://no-yellow-node"),
+		},
+	}
+	chainCfg, nodePool, nodes, err := client.NewClientConfigs(ptr("HighestHead"), time.Duration(0), "", nodeConfigs,
+		ptr[uint32](5), 10*time.Second, ptr[uint32](5), ptr(false), time.Minute, ptr[uint32](5), ptr(false))
+	require.NoError(t, err)
+
+	chainID := big.NewInt(68472)
+	evmClient := client.NewEvmClient(nodePool, chainCfg, nil, logger.Test(t), chainID, nodes, "")
+
+	err = evmClient.Dial(ctx)
+	require.NoError(t, err)
+	defer evmClient.Close()
+
+	expectedMetrics := map[string]struct{}{
+		`evm_pool_rpc_node_dials_total{evmChainID="68472",nodeName="BlueEVMPrimaryNode"}`:   {},
+		`evm_pool_rpc_node_dials_total{evmChainID="68472",nodeName="YellowEVMPrimaryNode"}`: {},
+		`multi_node_states{chainId="68472",network="EVM",state="Alive"}`:                    {},
```

Comment:
nit: If someone adds a new state tomorrow, this test will be obsolete, unless they add that new state here too explicitly.
Isn't there anything in golang that can allow us to iterate and get a list of all enums to ensure this test covers all of them?

Or worst case, could we add a check here that
assert.Equal(t,nodeStateLen, 9)
as per this:
https://github.com/smartcontractkit/chainlink/blob/f8c448a83db09d4c08b0e77b5326f935bb4b21bd/common/client/node_fsm.go#L102

Thus if a new state gets added, this test assertion will ensure we update it here too.

Reply:
Good point. Unfortunately `nodeState` is a private type in `common/client` package, so we can not add checks that you've mentioned. 
But Dylan makes `nodeState` public in his Chain Agnostic MultiNode PR. I'll include the checks once his PR is merged.